File: sol/solution/0001-0100/0011. Container With Most Water/Solution.ts
function maxArea(height: number[]): number {
    let left: number = 0,
        right: number = height.length - 1;
    let maxArea: number = 0;
    while (left < right) {
        const width: number = right - left;
        const currentHeight: number = Math.min(height[left], height[right]);
        maxArea = Math.max(maxArea, width * currentHeight);
        if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }
    return maxArea;
}

File: sol/solution/0001-0100/0011. Container With Most Water/Solution.cs
public class Solution
{
    public int MaxArea(int[] height)
    {
        int left = 0, right = height.Length - 1;
        int maxArea = 0;
        while (left < right)
        {
            int width = right - left;
            int currentHeight = Math.Min(height[left], height[right]);
            maxArea = Math.Max(maxArea, width * currentHeight);
            if (height[left] < height[right])
            {
                left++;
            }
            else
            {
                right--;
            }
        }
        return maxArea;
    }
}